### ISSUE TITLE
Inconsistency in column-cursor position when creating/undoing a new column

### ISSUE THREAD

**Issue by jsvine** [saulpw/visidata#1445]
**Small description**

VisiData provides various ways to create new and derived columns, including but not limited to: `addcol-expr`, `addcol-incr`, and `addcol-new`. But VisiData's column-cursor's behavior is inconsistent when creating/undoing these commands. Using the examples of the three aforementioned commands, executing them while on the first column of a freshly-loaded `vd sample_data/benchmark.csv`:

- `addcol-expr` (`=`, with `len(Date)` as the input): Creates the new column directly to the right of the cursor column, and moves the cursor to that new column. This seems like the expected behavior.
- `addcol-incr` (`i`): Same as above.
- `addcol-new` (`za`): Creates a new column directly to the right of the cursor column ... but then moves the cursor *two* columns to the right, so that it's highlighting the column after the newly-created one. This seems unexpected / a bug, and seems caused by a combination of https://github.com/saulpw/visidata/commit/7bc2fb398458a305a3662e71a50f2150f2354530 (new to v2.9) and the call to `cursorRight(1)` in `addcol-new`'s definition [here](https://github.com/saulpw/visidata/blob/e2bbb7f60e069297f5da86f1cbe6faf3f606f72e/visidata/modify.py#L286).

When *undoing* (`U`) those commands:

- `addcol-expr`: Cursor moves back to the first column. This seems like the expected behavior.
- `addcol-incr`: Cursor moves *forward*, to the next column to the right. This seems like unexpected behavior.
- `addcol-new`: Cursor moves *even further forward* (given that it was already one column extra to the right) to the third column. This also seems like unexpected behavior.

Looking at the differences in those commands' definitions, it seems that the unexpected behavior by the latter two commands may stem from them not setting a `col` value in the corresponding cmdlog, which comes into play in `cmdlog.moveToReplayContext(...)`:

https://github.com/saulpw/visidata/blob/647ccf076cc1457b6ad5286580ead85794913571/visidata/cmdlog.py#L286-L292

... which itself is called by `undo.undo(...)`:

https://github.com/saulpw/visidata/blob/647ccf076cc1457b6ad5286580ead85794913571/visidata/undo.py#L35-L51

Things seem to work fine for`addcol-expr` due to it setting `curcol=cursorCol` in its definition:

https://github.com/saulpw/visidata/blob/647ccf076cc1457b6ad5286580ead85794913571/visidata/expr.py#L54

Tweaking `addcol-incr` so that the relevant part becomes `SettableColumn(type=int, curcol=cursorCol)` *seems* to do the trick, but I'm not familiar enough with the codebase to know whether that's the best approach — or whether it's the most generalize-able for the purpose of achieving cross-command consistency.

**Expected result**

See above.

**Actual result with screenshot**

See above.

**Steps to reproduce with sample data and a .vd**

See above.

**Additional context**

VisiData v2.9 and/or `develop` branch.

**Comment by anjakefala** [comment]
This is such a well written issue, @jsvine!

**Comment by saulpw** [comment]
Thanks for reporting the issue and for doing all the legwork, @jsvine!  I was able to make short work of this bug because of it.  You were completely right in your analysis; I chose to add `'ColumnAtCursor'` to the list of things to check the execstr for, as the most general solution, instead of tweaking individual commands to add `cursorCol` to them.  This whole cmdlog/undo setup is a glorious hack that may not be theoretically sound, but seems to work well for what it does, and so we'll keep it around for another day.
